### PR TITLE
Install MPI by default; always check_env; consistent tab formatting

### PR DESCRIPTION
**Pull request by jwallwork23**
I was having an issue running MPI parallel tests locally because of conflicting MPI builds.

This PR changes the installation process so that existing MPI builds are not picked up by default. It also ensures that the `check_env` command is always run.

### CODE REVIEW

**Comment by ddundo** [comment]
I thought about what we discussed during the meeting about skipping `check_env` during `make install`. Can't a user just do `yes | make install` to skip all prompts?

**Comment by jwallwork23** [comment]
> I thought about what we discussed during the meeting about skipping `check_env` during `make install`. Can't a user just do `yes | make install` to skip all prompts?

True, although I think that's a little less clear than having a specific recipe.

**Review by ddundo — CHANGES_REQUESTED** [review]
I tested this with #77 and it looks good, apart from one change in the Makefile. Although the cache errors are hit there as well in Goalie

File: install/Makefile [hunk 1]
```diff
@@ -75,12 +77,16 @@ check_env:
...
-install:
+# Install approach that doesn't require user input
+install_no_chk:
 	@cd $(BUILD_DIR); \
 		export PETSC_CONFIGURE_OPTIONS=$(PETSC_CONFIGURE_OPTIONS); \
 		curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install; \
 		python3 firedrake-install $(FIREDRAKE_CONFIGURE_OPTIONS)
 
+# Standard install approach that asks user to check environment setup
+install: check_env
```
[the remark this answers is not in the document]
Comment (jwallwork23) [reply install/Makefile:88]: Oops! I think what I was thinking of was this:
```suggestion
install: check_env install_no_chk
```

[the remark this answers is not in the document]
Comment (ddundo) [reply install/Makefile:88]: Nice! Makes sense :)

File: .github/workflows/test_firedrake_install.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  # Trigger the workflow on the 1st day of every month at midnight
+  schedule:
+    - cron: '0 0 1 * *'
+
+  # Manually trigger the workflow
+  workflow_dispatch:
+
+# Stop the workflow if a new run is requested
+concurrency:
+  group: 'install_firedrake_test-${{ github.ref }}'
```
Comment (ddundo) [inline .github/workflows/test_firedrake_install.yml:19]:
```suggestion
  group: 'test_firedrake_install-${{ github.ref }}'
```
Forgot to rename this to match filename, sorry

File: .github/workflows/test_firedrake_install.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          openssh-client build-essential autoconf automake cmake gfortran git \
+          libopenblas-serial-dev libtool python3-dev python3-pip python3-tk \
+          python3-venv python3-requests zlib1g-dev libboost-dev sudo gmsh bison flex \
+          ninja-build libocct-ocaf-dev libocct-data-exchange-dev swig graphviz \
+          libcurl4-openssl-dev libxml2-dev
+
+      - name: 'Download necessary files'
+        run: |
+          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/fix-mpi-install/install/Makefile
+          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/fix-mpi-install/install/petsc_configure_options.txt
```
Comment (ddundo) [inline .github/workflows/test_firedrake_install.yml:45]:
```suggestion
          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/main/install/Makefile
          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/main/install/petsc_configure_options.txt
```
Change to main branch before merging

**Review by ddundo — APPROVED** [review]

**Pull request merged**